src/main/java/com/google/devtools/build/lib/skyframe/SequencedSkyframeExecutor.java

```diff
@@ -215,7 +215,11 @@ protected MemoizingEvaluator createEvaluator(
         recordingDiffer,
         progressReceiver,
         inconsistencyReceiver,
-        trackIncrementalState ? DEFAULT_EVENT_FILTER_WITH_ACTIONS : EventFilter.NO_STORAGE,
+        trackIncrementalState
+            ? (shouldStoreTransitivePackagesInLoadingAndAnalysis()
```

Comment:
It is not clear without further context what the relationship is between storing transitive packages and filtering external repo events. Suggestion for `SkyframeExecutor`:

```java
final boolean externalRepositoriesEnabled() {
  return repositoryHelpersHolder != null
}

private boolean shouldStoreTransitivePackagesInLoadingAndAnalysis() {
  return externalRepositoriesEnabled();
}
```

And then here branch on `externalRepositoriesEnabled()`.